### PR TITLE
Add nuclear weapons proliferation data

### PR DESCRIPTION
**Pull request by pabloarosado**

### CODE REVIEW

File: snapshots/war/2024-01-11/nuclear_threat_initiative_overview.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import click
+import pandas as pd
+
+from etl.snapshot import Snapshot
+
+# Version for current snapshot dataset.
+SNAPSHOT_VERSION = Path(__file__).parent.name
+
+# Latest year for which we have assume NTI has data.
+LATEST_YEAR = 2022
```
Comment (paarriagadap) [inline snapshots/war/2024-01-11/nuclear_threat_initiative_overview.py:21]: Should that year be 2023 now?

File: etl/steps/data/garden/war/2024-01-11/nuclear_threat_initiative_overview.meta.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+dataset:
+  update_period_days: 365
+
+tables:
+  nuclear_threat_initiative_overview:
+    variables:
+      status:
+        title: Nuclear weapons status
+        description_short: Whether a country does not consider (0), considers (1), pursues (2), or possesses (3) nuclear weapons.
+        processing_level: major
```
Comment (paarriagadap) [inline etl/steps/data/garden/war/2024-01-11/nuclear_threat_initiative_overview.meta.yml:10]: Again, it would be cool to add a `description_processing` since the processing level is major

**Review by paarriagadap — APPROVED** [review]
Very good! I added minor comments

File: etl/steps/data/meadow/war/2024-01-11/spread_of_nuclear_weapons.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    # Load inputs.
+    #
+    # Retrieve snapshot.
+    snap = paths.load_snapshot("spread_of_nuclear_weapons.csv")
+
+    #
+    # Process data.
+    #
+    # The raw data is in a table in a PDF document, which is hard to parse programmatically.
+    # Instead, we manually extract the values here.
```
[the remark this answers is not in the document]
Comment (pabloarosado) [reply etl/steps/data/meadow/war/2024-01-11/spread_of_nuclear_weapons.py:20]: Good point, it's incorrect to add the snapshot as a csv file when it's a pdf. I've corrected that. However, in this case I think it's more useful if the snapshot is the pdf, which is the real source of truth (that we can visually check), rather than the manually extracted data (which could have errors due to the manual extraction).

File: etl/steps/data/garden/war/2024-01-11/spread_of_nuclear_weapons.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+from owid.catalog import Table
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+# Latest year to be assumed for the content of the data, when intervals are open, e.g. "2000-", or "1980-".
+LATEST_YEAR = 2017
```
[the remark this answers is not in the document]
Comment (pabloarosado) [reply etl/steps/data/garden/war/2024-01-11/spread_of_nuclear_weapons.py:12]: Good point.

File: etl/steps/data/garden/war/2024-01-11/spread_of_nuclear_weapons.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+            # A country that considers nuclear weapons must not be coded as possessing or pursuing nuclear weapons.
+            assert (year not in row["pursue"]) and (year not in row["acquire"])
+
+    # Add metadata to the new status column.
+    tb["status"] = tb["status"].copy_metadata(tb["explore"])
+
+    return tb
+
+
+def run(dest_dir: str) -> None:
```
[the remark this answers is not in the document]
Comment (pabloarosado) [reply etl/steps/data/garden/war/2024-01-11/spread_of_nuclear_weapons.py:98]: Done.

File: etl/steps/data/garden/war/2024-01-11/spread_of_nuclear_weapons.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        tb[tb["country"] == "Germany"]["explore"].astype(str).item()
+        + ","
+        + tb[tb["country"] == "West Germany"]["explore"].astype(str).item()
+    )
+    # Remove row for West Germany.
+    tb = tb[tb["country"] != "West Germany"].reset_index(drop=True)
+
+    # Assign the years of nuclear weapons exploration and pursuit of Yugoslavia to Serbia.
+    # To do that, simply rename the country.
+    tb.loc[tb["country"] == "Yugoslavia", "country"] = "Serbia"
```
[the remark this answers is not in the document]
Comment (pabloarosado) [reply etl/steps/data/garden/war/2024-01-11/spread_of_nuclear_weapons.py:51]: I agree this one is not very clear. I did this, firstly, because that's how it was done in the current version in grapher (so I assume Bastian put some thought on that already), and secondly, because it would be inconsistent to keep Yugoslavia when there are no other historical regions (e.g. there is no USSR). But Bastian and I are currently discussing whether we should include historical regions.

File: snapshots/war/2024-01-11/nuclear_threat_initiative_overview.py (hunk 1 above)
Comment (pabloarosado) [reply snapshots/war/2024-01-11/nuclear_threat_initiative_overview.py:21]: Yes, Bastian has confirmed that 2023 is also fully informed by the NTI.

File: etl/steps/data/garden/war/2024-01-11/nuclear_threat_initiative_overview.meta.yml (hunk 2 above)
Comment (pabloarosado) [reply etl/steps/data/garden/war/2024-01-11/nuclear_threat_initiative_overview.meta.yml:10]: Done.

**Comment by pabloarosado** [comment]
Thanks for the suggestions, @paarriagadap!
